x-pack/legacy/plugins/apm/server/routes/errors.ts

```diff
@@ -4,88 +4,75 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Boom from 'boom';
-import Joi from 'joi';
-import { InternalCoreSetup } from 'src/core/server';
+import * as t from 'io-ts';
+import { createRoute } from './create_route';
 import { getErrorDistribution } from '../lib/errors/distribution/get_distribution';
 import { getErrorGroup } from '../lib/errors/get_error_group';
 import { getErrorGroups } from '../lib/errors/get_error_groups';
-import { withDefaultValidators } from '../lib/helpers/input_validation';
 import { setupRequest } from '../lib/helpers/setup_request';
+import { uiQueryRt } from './default_api_types';
 
-const defaultErrorHandler = (err: Error) => {
-  // eslint-disable-next-line
-  console.error(err.stack);
-  throw Boom.boomify(err, { statusCode: 400 });
-};
+export const errorsRoute = createRoute(core => ({
+  path: '/api/apm/services/{serviceName}/errors',
+  params: {
+    path: t.type({
+      serviceName: t.string
+    }),
+    query: t.intersection([
+      t.partial({
+        sortField: t.string,
+        sortDirection: t.string
+      }),
+      uiQueryRt
```

Comment:
`uiQueryRt` is an odd concept and not possible to understand without going to definition. I still think range and uiFilter should be separate. 
Wrt `_debug` can we make that a default part of all routes. So it's not necessary to add explicitly?

Reply:
Sure on both counts. I opted for `uiQueryRt` because creating intersection types with `io-ts` is kind of annoying. I'll replace, and I'll try to include `minimalRt` by default. I might run into some type issues though, will have to investigate.

Reply:
Thanks!

Reply:
_debug is in the route defaults now, and I've separated `uiQueryRt` into `rangeRt` and `uiFiltersRt`